### PR TITLE
Avoid using mavenLocal() in build.gradle?

### PR DESCRIPTION
**Pull request by nickshoe**
##### **Overview of the feature request**

While reading the Gradle documentation to configure Gradle to use my private Nexus repository, I ran into this advice in the [official documentation](https://docs.gradle.org/current/userguide/declaring_repositories.html#sec:case-for-maven-local) regarding maven repositories:

> As a general advice, you should avoid adding `mavenLocal()` as a repository...

Existing code: 

https://github.com/jhipster/generator-jhipster/blob/1b76ae1b002eaf9498e9cbc190fb675de991624f/generators/server/templates/build.gradle.ejs#L283-L285

##### **Motivation for or Use Case**

As stated in the doc's chapter ["Obtaining module metadata"](https://docs.gradle.org/current/userguide/dependency_resolution.html#obtaining_module_metadata):

> - Each repository is inspected in order.
>   - Depending on the type of repository, Gradle looks for metadata files describing the module (`.module`, `.pom` or `ivy.xml` file) or directly for artifact files.
>   - Modules that have a module metadata file (`.module`, `.pom` or `ivy.xml` file) are preferred over modules that have an artifact file only.
>   - Once a repository returns a _metadata_ result, following repositories are ignored.

(penultimate point)
> - All of the artifacts for the module are then requested from the same repository that was chosen in the process above.

(callout)
> The penultimate point above is what can make the integration with Maven Local problematic. As it is a cache for Maven, it will sometimes miss some artifacts of a given module. If Gradle is sourcing such a module from Maven Local, it will consider the missing artifacts to be missing altogether.

##### **Related issues or PR**

- [x] Checking this box is mandatory (this is just to show you read everything)

### CODE REVIEW

**Comment by mshima** [comment]
JHipster uses `mavenLocal()` for jhipster-bom at CI.
Would require us to start using snapshots (https://github.com/orgs/jhipster/packages?repo_name=jhipster-bom) instead.

**Comment by nickshoe** [comment]
Hey @mshima I'm new to CI concepts and practices.. ':D
Can you point me to an explanation/tutorial of what jhipster-bom and snapshots are?
Let me know if I can do something to this PR.
Thanks

**Comment by atomfrede** [comment]
Ah you are right @mshima . Didn't remember why we kept it.

File: generators/server/templates/build.gradle.ejs [hunk 1]
```diff
@@ -281,7 +281,6 @@ configurations {
 }
 
 repositories {
-    mavenLocal()
     mavenCentral()
```
Comment (mshima) [inline generators/server/templates/build.gradle.ejs:284]: We can make it optional on jhipster-bom snapshot.
```suggestion
    // Local maven repository is required for libraries built locally with maven like development jhipster-bom.
    <%= !JHIPSTER_DEPENDENCIES_VERSION.includes('-SNAPSHOT') ? '// ' : '' %>mavenLocal()
    mavenCentral()
```

**Comment by mshima** [comment]
@atomfrede what do you think about this now?

**Review by atomfrede — APPROVED** [review]

**Comment by atomfrede** [comment]
Sorry missed the review request.

**Pull request merged**